app/scripts/lib/stores/persistence-manager.ts

```diff
@@ -1,9 +1,49 @@
+/**
+ * navigator.locks is a polyfill for the Locks API, which can be removed from
+ * this file after we drop support for older Firefox versions.
+ * navigator.locks is supported in Firefox 102+.
+ */
+import 'navigator.locks';
 import log from 'loglevel';
 import { captureException } from '@sentry/browser';
 import { isEmpty } from 'lodash';
-import { type MetaMaskStateType, MetaMaskStorageStructure } from './base-store';
-import ExtensionStore from './extension-store';
-import ReadOnlyNetworkStore from './read-only-network-store';
+import { KeyringControllerState } from '@metamask/keyring-controller';
+import { MISSING_VAULT_ERROR } from '../../../../shared/constants/errors';
+import { OnboardingControllerState } from '../../controllers/onboarding';
+import { PreferencesControllerState } from '../../controllers/preferences-controller';
+import { AppStateControllerState } from '../../controllers/app-state-controller';
+import { IndexedDBStore } from './indexeddb-store';
+import type {
+  MetaMaskStateType,
+  MetaMaskStorageStructure,
+  BaseStore,
+  MetaData,
+} from './base-store';
+
+export type Backup = {
+  KeyringController?: KeyringControllerState;
+  OnboardingController?: OnboardingControllerState;
+  PreferencesController?: PreferencesControllerState;
+  AppStateController?: AppStateControllerState;
+};
+
+function makeBackup(state?: MetaMaskStateType): Backup {
+  return {
+    KeyringController: state?.KeyringController,
+    OnboardingController: state?.OnboardingController,
+    PreferencesController: state?.PreferencesController,
+    AppStateController: state?.AppStateController,
+  } as Backup;
```

Comment:
Hmm, this doesn't seem correctly typed. We don't know that the vault backups match this state shape. Even putting aside the "corrupted state" case, it might be from a previous wallet version.

Could we type `Backup` as having properties of type `unknown` instead? That would be a "more correct" representation of what we know about the state, and it avoids the need to cast.

I see that would be generally OK for this PR, but I'm not sure if that would cause problems for future PRs.

Reply:
fixed